src/core/blob.cpp

```diff
@@ -14,51 +14,76 @@
 
 using namespace soci;
 
+#define SOCI_BLOB_ENSURE_INITIALIZED \
+    if (!is_valid()) \
+    { \
+        throw soci_error("Attempted to access invalid blob"); \
+    }
+
```

Comment:
There doesn't seem to be any good reason to make this a macro, could we please make it a (member) function instead?

Reply:
The only reason is that I didn't want this implementation detail to leak into the public header (which it will if we make it a member function - even if that function is private).
But I don't mind making this a member function either :shrug: